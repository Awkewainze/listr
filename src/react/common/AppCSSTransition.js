import React from 'react';
import CSSTransition from 'react-transition-group/cjs/CSSTransition';

const generateClassNames = (prefix = '') => ({
  appear: `${prefix}--will-appear`,
  appearActive: `${prefix}--appearing`,
  appearDone: `${prefix}--appeared`,
  enter: `${prefix}--will-enter`,
  enterActive: `${prefix}--entering`,
  enterDone: `${prefix}--entered`,
  exit: `${prefix}--will-exit`,
  exitActive: `${prefix}--exiting`,
  exitDone: `${prefix}--exited`
});

const AppCSSTransition = ({
  prefix,
  timeout,
  children,
  ...props
}) => {
  const isIn = props.in;

  return (
    <CSSTransition in={isIn} timeout={timeout} classNames={generateClassNames(prefix)} {...props} >
      {children}
    </CSSTransition>
  );
};

export default AppCSSTransition;