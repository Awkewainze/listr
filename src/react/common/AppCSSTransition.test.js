import React from 'react';
import { shallow } from 'enzyme';
import AppCSSTransition from './AppCSSTransition';
import { CSSTransition } from 'react-transition-group';

const TEST_PROPS = {
  prefix: 'testPrefix',
  timeout: 50
};

it('matches snapshot', () => {
  expect(shallow(<AppCSSTransition {...TEST_PROPS}><p>Child</p></AppCSSTransition>)).toMatchSnapshot();
});

it('correctly configures classNames', () => {
  const wrapper = shallow(<AppCSSTransition {...TEST_PROPS} ><p>Child</p></AppCSSTransition>);
  const classNames = wrapper.find(CSSTransition).prop('classNames');
  Object.values(classNames).forEach(className => {
    expect(className.startsWith(TEST_PROPS.prefix)).toBeTruthy();
  });
});