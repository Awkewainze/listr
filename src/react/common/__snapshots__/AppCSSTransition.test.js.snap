// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<CSSTransition
  classNames={
    Object {
      "appear": "testPrefix--will-appear",
      "appearActive": "testPrefix--appearing",
      "appearDone": "testPrefix--appeared",
      "enter": "testPrefix--will-enter",
      "enterActive": "testPrefix--entering",
      "enterDone": "testPrefix--entered",
      "exit": "testPrefix--will-exit",
      "exitActive": "testPrefix--exiting",
      "exitDone": "testPrefix--exited",
    }
  }
  timeout={50}
>
  <p>
    Child
  </p>
</CSSTransition>
`;
